src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -86,6 +86,14 @@ private static function prepareRequest(?string $method, ?string $url, array $opt
 
         if (isset($options['body'])) {
             $options['body'] = self::normalizeBody($options['body']);
+
+            if (\is_string($options['body'])
+                && (string) \strlen($options['body']) !== substr($h = $options['normalized_headers']['content-length'][0] ?? '', 16)
+                && ('' !== $h || ('' !== $options['body'] && !isset($options['normalized_headers']['transfer-encoding'])))
+            ) {
+                $options['normalized_headers']['content-length'] = [substr_replace($h ?: 'Content-Length: ', \strlen($options['body']), 16)];
+                $options['headers'] = array_merge(...array_values($options['normalized_headers']) ?: [[]]);
```

Comment:
Can `array_values($options['normalized_headers'])` be falsy while we just set `'content-length'` above?

Reply:
'headers' is a list, we need to remove the previous content-length and see the new one. This does it.

Reply:
got it, updated